components/renku_data_services/notebooks/blueprints.py

```diff
@@ -269,59 +288,67 @@ async def _handler(
-                    configuration=s.configuration,
-                    readonly=s.readonly,
+            data_connectors_stream = self.data_connector_secret_repo.get_data_connectors_with_secrets(user, project.id)
+            dcs: dict[str, RCloneStorage] = {}
+            dcs_secrets: dict[str, list[DataConnectorSecret]] = {}
+            async for dc in data_connectors_stream:
+                dcs[str(dc.data_connector.id)] = RCloneStorage(
+                    source_path=dc.data_connector.storage.source_path,
+                    mount_folder=dc.data_connector.storage.target_path
+                    if PurePosixPath(dc.data_connector.storage.target_path).is_absolute()
+                    else (work_dir / dc.data_connector.storage.target_path).as_posix(),
+                    configuration=dc.data_connector.storage.configuration,
+                    readonly=dc.data_connector.storage.readonly,
                     config=self.nb_config,
-                    name=None,
+                    name=dc.data_connector.name,
                 )
-                for s in body.cloudstorage or []
-            }
+                if len(dc.secrets) > 0:
+                    dcs_secrets[str(dc.data_connector.id)] = dc.secrets
             # NOTE: Check the cloud storage in the request body and if any match
             # then overwrite the projects cloud storages
-            # NOTE: Cloud storages in the session launch request body that are not form the DB will cause a 422 error
-            for csr_id, csr in cloud_storage_request.items():
-                if csr_id not in cloud_storage:
+            # NOTE: Cloud storages in the session launch request body that are not from the DB will cause a 404 error
+            # NOTE: Overriding the configuration when a saved secret is there will cause a 422 error
+            cloud_storage_overrides = body.cloudstorage or []
+            for csr in cloud_storage_overrides:
+                csr_id = csr.storage_id
+                if csr_id not in dcs:
                     raise errors.MissingResourceError(
                         message=f"You have requested a cloud storage with ID {csr_id} which does not exist "
                         "or you dont have access to.",
                         quiet=True,
                     )
-                cloud_storage[csr_id] = csr
+                if csr.target_path is not None and not PurePosixPath(csr.target_path).is_absolute():
+                    csr.target_path = (work_dir / csr.target_path).as_posix()
+                # TODO: The UI always does this check if it is acceptable/safe
+                # if csr_id in dcs_secrets and csr.configuration is not None:
+                #     raise errors.ValidationError(
+                #         message=f"Overriding the storage configuration for storage with ID {csr_id} "
+                #         "is not allowed because the storage has an associated saved secret.",
+                #     )
```

Comment:
Note: I am not sure about the validity of this comment. What if both `access_key_id` and `secret_access_key` are required, but I only saved `access_key_id`?

Reply:
Yes I meant to remove this. I will remove it now.

Reply:
Yes I meant to remove this before requesting a review but I forgot.